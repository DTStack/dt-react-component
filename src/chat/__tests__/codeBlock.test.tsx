import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import CodeBlock, { ICodeBlockProps } from '../codeBlock';

function Container({
    language,
    children,
    ...rest
}: { language: string; children: React.ReactNode } & Omit<ICodeBlockProps, 'children'>) {
    return (
        <CodeBlock {...rest}>
            <span className={`language-${language}`}>{children}</span>
            <></>
        </CodeBlock>
    );
}

describe('Test Chat CodeBlock', () => {
    beforeEach(cleanup);

    it('Match snapshot', () => {
        expect(
            render(
                <Container language="javascript" className="jest_test" style={{ color: 'red' }}>
                    const a = 1
                </Container>
            ).asFragment()
        ).toMatchSnapshot('normal');

        expect(
            render(
                <Container language="javascript" convert>
                    const a = 1
                </Container>
            ).asFragment()
        ).toMatchSnapshot('normal');
    });

    it('Should support configure copy', () => {
        const { container, getByTestId, rerender } = render(
            <Container
                language="javascript"
                copy={{ text: '测试', button: <button data-testid="copyButton">测试</button> }}
            >
                const a = 1
            </Container>
        );
        expect(getByTestId('copyButton')).toBeInTheDocument();

        rerender(
            <Container language="javascript" copy={false}>
                const a = 1
            </Container>
        );
        expect(container.querySelector('.dtc-copy')).toBeNull();
    });
});
