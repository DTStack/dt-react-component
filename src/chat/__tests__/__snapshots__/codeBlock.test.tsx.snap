// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Chat CodeBlock Match snapshot: normal 1`] = `
<DocumentFragment>
  <div
    class="dtc__aigc__codeblock jest_test"
    style="color: red;"
  >
    <div
      class="dtc__aigc__codeblock__header"
    >
      <span
        class="dtc__aigc__codeblock__language"
      >
        javascript
      </span>
      <div
        class="dtc__aigc__codeblock__tool"
      >
        <span
          class="dtc-copy"
        >
          <span
            class="dtc__icon"
          >
            <svg
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                stroke="none"
                stroke-width="1"
              >
                <g
                  transform="translate(-340 -367)"
                >
                  <g
                    transform="translate(201 320)"
                  >
                    <g
                      transform="translate(139 47)"
                    >
                      <path
                        d="M11 1.5A1.5 1.5 0 0112.5 3v.25h.5a1.5 1.5 0 011.5 1.5v8a1.5 1.5 0 01-1.5 1.5H5a1.5 1.5 0 01-1.5-1.5v-.25H3a1.5 1.5 0 01-1.493-1.356L1.5 11V3A1.5 1.5 0 013 1.5zm1.5 9.5a1.5 1.5 0 01-1.5 1.5H4.5v.25a.5.5 0 00.5.5h8a.5.5 0 00.5-.5v-8a.5.5 0 00-.5-.5h-.5zM11 2.5H3a.5.5 0 00-.5.5v8a.5.5 0 00.5.5h8a.5.5 0 00.5-.5V3a.5.5 0 00-.5-.5zm-4 1a.5.5 0 01.5.5v2.5H10a.5.5 0 110 1H7.5V10a.5.5 0 11-1 0V7.5H4a.5.5 0 010-1h2.5V4a.5.5 0 01.5-.5z"
                        fill="currentColor"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </span>
        </span>
      </div>
    </div>
    <pre
      style="background: rgb(255, 255, 255); color: rgb(56, 56, 56); font-family: "Fira Code", "Fira Mono", Menlo, Consolas, "DejaVu Sans Mono", monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 2; hyphens: none; padding: 1em; margin: 0px; overflow: auto; border-radius: 0 0 4px 4px; max-height: 270px;"
    >
      <code
        class="language-javascript"
        style="background: rgb(255, 255, 255); color: rgb(56, 56, 56); font-family: "Fira Code", "Fira Mono", Menlo, Consolas, "DejaVu Sans Mono", monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 2; hyphens: none;"
      >
        <span>
          <span
            class="linenumber react-syntax-highlighter-line-number"
            style="display: inline-block; min-width: 1.25em; padding-right: 1em; text-align: right; user-select: none; font-style: italic; color: rgb(160, 160, 160);"
          >
            1
          </span>
          <span
            class="token"
            style="color: rgb(38, 38, 38);"
          >
            const
          </span>
          <span>
             a 
          </span>
          <span
            class="token"
            style="color: rgb(64, 64, 64);"
          >
            =
          </span>
          <span>
             
          </span>
          <span
            class="token"
            style="color: rgb(1, 1, 1);"
          >
            1
          </span>
        </span>
      </code>
    </pre>
  </div>
</DocumentFragment>
`;

exports[`Test Chat CodeBlock Match snapshot: normal 2`] = `
<DocumentFragment>
  <div
    class="dtc__aigc__codeblock dtc__aigc__codeblock--convert"
  >
    <div
      class="dtc__aigc__codeblock__header"
    >
      <span
        class="dtc__aigc__codeblock__language"
      >
        javascript
      </span>
      <div
        class="dtc__aigc__codeblock__tool"
      >
        <span
          class="dtc-copy"
        >
          <span
            class="dtc__icon"
          >
            <svg
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                stroke="none"
                stroke-width="1"
              >
                <g
                  transform="translate(-340 -367)"
                >
                  <g
                    transform="translate(201 320)"
                  >
                    <g
                      transform="translate(139 47)"
                    >
                      <path
                        d="M11 1.5A1.5 1.5 0 0112.5 3v.25h.5a1.5 1.5 0 011.5 1.5v8a1.5 1.5 0 01-1.5 1.5H5a1.5 1.5 0 01-1.5-1.5v-.25H3a1.5 1.5 0 01-1.493-1.356L1.5 11V3A1.5 1.5 0 013 1.5zm1.5 9.5a1.5 1.5 0 01-1.5 1.5H4.5v.25a.5.5 0 00.5.5h8a.5.5 0 00.5-.5v-8a.5.5 0 00-.5-.5h-.5zM11 2.5H3a.5.5 0 00-.5.5v8a.5.5 0 00.5.5h8a.5.5 0 00.5-.5V3a.5.5 0 00-.5-.5zm-4 1a.5.5 0 01.5.5v2.5H10a.5.5 0 110 1H7.5V10a.5.5 0 11-1 0V7.5H4a.5.5 0 010-1h2.5V4a.5.5 0 01.5-.5z"
                        fill="currentColor"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </span>
        </span>
      </div>
    </div>
    <pre
      style="background: rgb(249, 249, 250); color: rgb(56, 56, 56); font-family: "Fira Code", "Fira Mono", Menlo, Consolas, "DejaVu Sans Mono", monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 2; hyphens: none; padding: 1em; margin: 0px; overflow: auto; border-radius: 0 0 4px 4px; max-height: 270px;"
    >
      <code
        class="language-javascript"
        style="background: rgb(249, 249, 250); color: rgb(56, 56, 56); font-family: "Fira Code", "Fira Mono", Menlo, Consolas, "DejaVu Sans Mono", monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 2; hyphens: none;"
      >
        <span>
          <span
            class="linenumber react-syntax-highlighter-line-number"
            style="display: inline-block; min-width: 1.25em; padding-right: 1em; text-align: right; user-select: none; font-style: italic; color: rgb(160, 160, 160);"
          >
            1
          </span>
          <span
            class="token"
            style="color: rgb(38, 38, 38);"
          >
            const
          </span>
          <span>
             a 
          </span>
          <span
            class="token"
            style="color: rgb(64, 64, 64);"
          >
            =
          </span>
          <span>
             
          </span>
          <span
            class="token"
            style="color: rgb(1, 1, 1);"
          >
            1
          </span>
        </span>
      </code>
    </pre>
  </div>
</DocumentFragment>
`;
